affordance_primitives/test/test_task_monitor.cpp

```diff
@@ -0,0 +1,194 @@
+//          defects in software and/or documentation, or loss or inaccuracy of
+//          data of any kind.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+
+#include <affordance_primitives/task_monitor/task_monitor.hpp>
+#include <affordance_primitives/msg_types.hpp>
+
+#include <thread>
+
+// constexpr double EPSILON = 1e-4;
+
+using affordance_primitives::ExecutionResult;
+
+TEST(TaskMonitor, start_stop_monitor)
+{
+  ros::NodeHandle nh;
+
+  // Default parameters
+  affordance_primitives::AffordanceParameter params;
+
+  // Create and start monitor with no timeout
+  affordance_primitives::TaskMonitor monitor(nh, "ft_topic_name");
+  ASSERT_NO_THROW(monitor.startMonitor(params));
+
+  // Wait a sec, then stop monitor
+  ros::Duration(0.25).sleep();
+  ASSERT_NO_THROW(monitor.stopMonitor());
+
+  // Result should be STOP_REQUESTED
+  std::optional<ExecutionResult> result = monitor.getResult();
+  ASSERT_TRUE(result.has_value());
+  EXPECT_EQ(result.value(), affordance_primitives::ExecutionResult::STOP_REQUESTED);
+
+  // Now try interupting a timed run
+  ASSERT_NO_THROW(monitor.startMonitor(params, 0.25));
+  ASSERT_NO_THROW(monitor.stopMonitor());
+  result = monitor.getResult();
+  ASSERT_TRUE(result.has_value());
+  EXPECT_EQ(result.value(), affordance_primitives::ExecutionResult::STOP_REQUESTED);
+}
+
+TEST(TaskMonitor, monitor_timing)
+{
+  ros::NodeHandle nh;
+  const double test_duration = 0.25;
+
+  // Default parameters
+  affordance_primitives::AffordanceParameter params;
+
+  // Create and start monitor with no timeout
+  affordance_primitives::TaskMonitor monitor(nh, "ft_topic_name");
+  ASSERT_NO_THROW(monitor.startMonitor(params, test_duration));
+
+  // Immediately getting the result shouldn't work
+  EXPECT_FALSE(monitor.getResult().has_value());
+
+  // Wait some time and grab result
+  ros::Duration(1.05 * test_duration).sleep();
+  std::optional<ExecutionResult> result = monitor.getResult();
+  ASSERT_TRUE(result.has_value());
+  EXPECT_EQ(result.value(), affordance_primitives::ExecutionResult::TIME_OUT);
+
+  // Now test blocking call to waitForResult()
+  ASSERT_NO_THROW(monitor.startMonitor(params, test_duration));
+  ExecutionResult result2;
+
+  ros::Time start = ros::Time::now();
+  result2 = monitor.waitForResult();
+  EXPECT_LE((ros::Time::now() - start).toSec(), 1.05 * test_duration);
+  EXPECT_EQ(result2, affordance_primitives::ExecutionResult::TIME_OUT);
+}
+
+TEST(TaskMonitor, monitor_ft_readings)
+{
+  ros::NodeHandle nh;
+  ros::Publisher ft_pub = nh.advertise<affordance_primitives::WrenchStamped>("ft_topic_name", 1, true);
+
+  const double test_duration = 0.25;
+  const double ft_pub_period = 0.005;  // 200 hz
+  const size_t num_publish = test_duration / ft_pub_period;
+
+  // Set some force limits in parameter
+  affordance_primitives::AffordanceParameter params;
```

Comment:
Are we going to move these params to a file for later? this question is based on Blake's comment on Chris's presentation last Friday (Feb 25th)

Reply:
These parameters are set on a per-task basis. Whatever will be calling the AP execution will need to set them. If they use a default in a file, that is up to them